lorawan-device/src/mac/multicast.rs

```diff
@@ -0,0 +1,188 @@
+use crate::mac;
+use crate::radio::RadioBuffer;
+use crate::Downlink;
+use lorawan::keys::{CryptoFactory, McKEKey};
+pub use lorawan::multicast::{self, Session};
+use lorawan::multicast::{
+    parse_downlink_multicast_messages, DownlinkRemoteSetup, McGroupSetupAnsCreator,
+};
+use lorawan::parser::FRMPayload;
+pub use lorawan::parser::McAddr;
+use lorawan::parser::{DataHeader, EncryptedDataPayload};
+
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
+pub enum Response {
+    NewSession(u8),
+    SessionExpired,
+    NoUpdate,
+    GroupSetupTransmitRequest(u8),
+    DownlinkReceived(u32),
+}
+
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt-03", derive(defmt::Format))]
+pub enum Error {}
+
+/// The port used for multicast setup message. The messages are "unicast" and encrypted & sent at
+/// the application layer.
+const REMOTE_MULTICAST_SETUP_PORT: u8 = 200;
+/// These ports are for actual multicast messages; they are encrypted and sent within a multicast
+/// session
+const DEFAULT_MC_PORT_START: u8 = 201;
```

Comment:
For future.. Could we use range here? Also, is the end inclusive? `(201..=205)` vs `(201..205)`

Reply:
Agree! Much nicer